src/main/java/io/appium/java_client/touch/OptionsCombinedWithOffset.java

```diff
@@ -0,0 +1,49 @@
+package io.appium.java_client.touch;
+
+import static java.util.Optional.ofNullable;
+
+import java.util.Map;
+
+public abstract class OptionsCombinedWithOffset<T extends OptionsCombinedWithOffset>
+        extends ActionOptions<OptionsCombinedWithOffset<T>> {
+    private ActionOptions<?> offsetOption;
+
+    /**
+     * Some actions may require some absolute offset value to be performed.
+     * Invocation of this method replaces the result of previous invocation of
+     * the {@link #withOffset(RelativeOffsetOption)}
+     *
+     * @param offset is the values of required absolute offset from the left corner of a screen.     *
+     * @return self-reference
+     */
+    public T withOffset(AbsoluteOffsetOption offset) {
+        offsetOption = offset;
+        return (T) this;
+    }
+
+    /**
+     * Some actions may require some relative offset value to be performed.
+     * Invocation of this method replaces the result of previous invocation of
+     * the {@link #withOffset(AbsoluteOffsetOption)}
+     *
+     * @param offset is the values of required offset from the left corner of an element.     *
+     * @return self-reference
+     */
+    public T withOffset(RelativeOffsetOption offset) {
+        offsetOption = offset;
+        return (T) this;
+    }
+
+    protected void verify() {
+        ofNullable(offsetOption).orElseThrow(() ->
```

Comment:
we should consider here, that useRelative(x,y) won't work as expected for press/tap/longTap. I think the best way to avoid it would be to extract element parameter from RelativeOffset into _withElement_ like we discussed before

Reply:
the main reason of this is that we anyway translate both relative and absolute offset into `x,y` JSON keys, so there is no way for the server do distinguish them (unless we start using [W3C JSON](https://github.com/jlipps/simple-wd-spec#perform-actions) format to represent the stuff, which is already awaiting in the queue).